api/middleware/cf_user.go

```diff
@@ -59,6 +59,9 @@ func (m *cfUser) middleware(next http.Handler) http.Handler {
 
 		if !isCFUser {
 			w.Header().Add("X-Cf-Warnings", fmt.Sprintf("Warning: subject '%s/%s' has no CF roles assigned. This is not supported and may cause unexpected behaviour.", identity.Kind, identity.Name))
```

Comment:
Do we still want to print this warning, given we're now returning an error? Maybe we should but the same message inside the error body?